src/main/java/cms/gov/madie/measure/validations/CqlObservationFunctionValidator.java

```diff
@@ -105,4 +105,44 @@ private Map<String, String> mapObservationsToValidPopBasis(String elmJson)
     }
     return observationPopBasis;
   }
+
+  public void validateObservationFunctionsForQdm(
+      Group group, String elmJson, boolean patientBasis, String cqlDefinitionReturnType)
+      throws JsonProcessingException {
+    Map<String, String> observationsToValidPopBasis = mapObservationsToValidPopBasis(elmJson);
+
+    List<MeasureObservation> observations = group.getMeasureObservations();
+
+    if (observationsToValidPopBasis.isEmpty()) {
+      if (!CollectionUtils.isEmpty(observations)) {
+        throw new InvalidMeasureObservationException(
+            "Measure CQL does not have observation definition");
+      } else {
+        return; // observations are optional for scoring types other than Continuous Variable.
+      }
+    }
+
+    if (observations != null) {
```

Comment:
can we not use CollectionUtils.isEmpty() instead of checking != null

Reply:
you might also want to do the same change for checking the stratifications in the 
src/main/java/cms/gov/madie/measure/validations/CqlDefinitionReturnTypeValidator.java --> line number 158

Reply:
@sb-prateekkeerthi Yeah sure I did a bunch.